rust/main/chains/hyperlane-sealevel/src/rpc/client.rs

```diff
@@ -245,19 +320,174 @@ impl SealevelRpcClient {
+            .units_consumed
+            .and_then(|units| if units > 0 { Some(units) } else { None })
+            .ok_or_else(|| {
+                ChainCommunicationError::from_other_str(
+                    "Empty or zero compute units returned in simulation result",
+                )
+            })?
+            .try_into()
+            .map_err(ChainCommunicationError::from_other)?;
+
+        // Bump the compute units by 10% to ensure we have enough, but cap it at the max.
+        let simulation_compute_units =
+            Self::MAX_COMPUTE_UNITS.min((simulation_compute_units * 11) / 10);
+
+        let priority_fee = priority_fee_oracle.get_priority_fee(&simulation_tx).await?;
+
+        Ok(SealevelTxCostEstimate {
+            compute_units: simulation_compute_units,
+            compute_unit_price_micro_lamports: priority_fee,
+        })
+    }
+
+    /// Builds a transaction with estimated costs for a given instruction.
+    pub async fn build_estimated_tx_for_instruction(
+        &self,
+        instruction: Instruction,
+        payer: &Keypair,
+        tx_submitter: &dyn TransactionSubmitter,
+        priority_fee_oracle: &dyn PriorityFeeOracle,
+    ) -> ChainResult<Transaction> {
+        // Get the estimated costs for the instruction.
+        let SealevelTxCostEstimate {
+            compute_units,
+            compute_unit_price_micro_lamports,
+        } = self
+            .get_estimated_costs_for_instruction(
+                instruction.clone(),
+                payer,
+                tx_submitter,
+                priority_fee_oracle,
+            )
+            .await?;
+
+        tracing::info!(
+            ?compute_units,
+            ?compute_unit_price_micro_lamports,
+            "Got compute units and compute unit price / priority fee for transaction"
+        );
+
+        // Build the final transaction with the correct compute unit limit and price.
+        let tx = self
+            .create_transaction_for_instruction(
+                compute_units,
+                compute_unit_price_micro_lamports,
+                instruction,
+                payer,
+                tx_submitter,
+                true,
+            )
+            .await?;
+
+        Ok(tx)
+    }
+
+    /// Creates a transaction for a given instruction, compute unit limit, and compute unit price.
+    /// If `sign` is true, the transaction will be signed.
+    pub async fn create_transaction_for_instruction(
+        &self,
+        compute_unit_limit: u32,
+        compute_unit_price_micro_lamports: u64,
+        instruction: Instruction,
+        payer: &Keypair,
+        tx_submitter: &dyn TransactionSubmitter,
+        sign: bool,
+    ) -> ChainResult<Transaction> {
+        let instructions = vec![
+            // Set the compute unit limit.
+            ComputeBudgetInstruction::set_compute_unit_limit(compute_unit_limit),
+            // Set the priority fee / tip
+            tx_submitter.get_priority_fee_instruction(
```

Comment:
nice, so this means you can't set both a priority fee and use jito, which would be ineffective 🔥

Reply:
yep! funnily enough Jito has some confusing docs where if you choose to use the non-MEV protected endpoint, they want you to use both priority fees and the jito tips. But we just won't support that for now to keep a nicer interface